VideoWeb/VideoWeb/Controllers/InternalEventController.cs

```diff
@@ -171,5 +177,25 @@ public async Task<IActionResult> AllocationHearings(AllocationHearingsToCsoReque
                 return StatusCode(e.StatusCode, e.Response);
             }
         }
+        
+        [HttpPost("HearingCancelled")]
+        [SwaggerOperation(OperationId = "HearingCancelled")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(string), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> HearingCancelled(Guid conferenceId)
+        {
+            await _hearingCancelledEventNotifier.PushHearingCancelledEvent(conferenceId);
```

Comment:
Remove the conference from the cache here?

Reply:
I don't think we can unfortunately. The base class for the event handlers (`EventHandlerBase`) fetches the conference from the cache. If we remove the conference, that step will fail

```
public virtual async Task HandleAsync(CallbackEvent callbackEvent)
        {
            SourceConference = await conferenceService.GetConference(callbackEvent.ConferenceId);
            if (SourceConference == null) throw new ConferenceNotFoundException(callbackEvent.ConferenceId);
```

Reply:
I didn't notice this was extending the EventHandlerBase class. The EventHandler base was designed to handler events from the supplier, not the internal events.
We've exposed internal events to suppliers if they were to send the correct payload to callback.

There is a ticket in the backlog to address VIH-9935, but ideally any internal event handlers shouldn't inherit from the design for the external event handler

Reply:
If you take this strategy to decouple the event handler, you can remove from the cache safely

Reply:
good idea. Have pushed 2 commits to address these points:

[First](https://github.com/hmcts/vh-video-web/pull/2219/commits/31eaab9cb67c783e94a02ab91c6b79c87d7f9610) - remove the event handlers for the new events I added, instead move the logic into the notifiers

[Second](https://github.com/hmcts/vh-video-web/pull/2219/commits/e8ba48801e153df2402b1a50b98bddb3fdcd823d) - remove the conference from the cache